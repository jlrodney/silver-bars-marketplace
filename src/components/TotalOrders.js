import React from 'react';
import IndividualOrder from './IndividualOrder';

const TotalOrders = (props) => (
  <div>
    <div> <p>All orders</p>
    {
      props.sellOrders.map((order) => (
        <IndividualOrder
          key={order.orderId}
          orderId={order.orderId}
          userId={order.userId}
          quantity={order.quantity}
          price={order.price}
          type={order.type}
          handleDeleteOrder={props.handleDeleteOrder}
        />
      ))
    }
    </div>
    <div>
    {
      props.buyOrders.map((order) => (
        <IndividualOrder
          key={order.orderId}
          orderId={order.orderId}
          userId={order.userId}
          quantity={order.quantity}
          price={order.price}
          type={order.type}
          handleDeleteOrder={props.handleDeleteOrder}
        />
      ))
    }
    </div>
  </div>
);

export default TotalOrders;
