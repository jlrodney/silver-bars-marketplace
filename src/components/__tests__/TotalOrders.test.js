import React from 'react';
import { shallow } from 'enzyme';
import TotalOrders from '../TotalOrders';

const buyOrders = [
  {
    orderId: 'xyz986',
    userId: '1',
    quantity: 22,
    price: 230,
    type: 'BUY'
  },
  {
    orderId: 'xyz987',
    userId: '3',
    quantity: 32,
    price: 2300,
    type: 'BUY'
  }
]

const sellOrders = [
  {
    orderId: 'xyz998',
    userId: '1',
    quantity: 222,
    price: 230,
    type: 'SELL'
  },
  {
    orderId: 'xyz989',
    userId: '2',
    quantity: 22,
    price: 230,
    type: 'SELL'
  }
]
describe('Orders component', () => {
  it('should render Orders with indivual  orders', () => {
    const wrapper = shallow(<TotalOrders buyOrders={buyOrders} sellOrders={sellOrders}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Orders with All orders header', () => {
    const wrapper = shallow(<TotalOrders buyOrders={buyOrders} sellOrders={sellOrders} />);
    expect(wrapper.find('p').text()).toEqual('All orders');
  });
});
