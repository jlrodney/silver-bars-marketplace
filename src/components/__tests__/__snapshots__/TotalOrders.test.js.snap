// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders component should render Orders with indivual  orders 1`] = `
<div>
  <div>
     
    <p>
      All orders
    </p>
    <IndividualOrder
      key="xyz998"
      orderId="xyz998"
      price={230}
      quantity={222}
      type="SELL"
      userId="1"
    />
    <IndividualOrder
      key="xyz989"
      orderId="xyz989"
      price={230}
      quantity={22}
      type="SELL"
      userId="2"
    />
  </div>
  <div>
    <IndividualOrder
      key="xyz986"
      orderId="xyz986"
      price={230}
      quantity={22}
      type="BUY"
      userId="1"
    />
    <IndividualOrder
      key="xyz987"
      orderId="xyz987"
      price={2300}
      quantity={32}
      type="BUY"
      userId="3"
    />
  </div>
</div>
`;
